mlir/lib/Analysis/AffineAnalysis.cpp

```diff
@@ -864,6 +410,41 @@ static void computeDirectionVector(
   }
 }
 
+LogicalResult MemRefAccess::getAccessRelation(FlatAffineRelation &ret) const {
+  // Create set corresponding to domain of access.
+  FlatAffineValueConstraints domain;
+  if (failed(getOpIndexSet(opInst, &domain)))
+    return failure();
+
+  // Get access relation from affine map.
+  AffineValueMap accessValueMap;
+  getAccessMap(&accessValueMap);
+  if (failed(getRelationFromMap(accessValueMap, ret)))
+    return failure();
+
+  FlatAffineRelation domainRel(ret.getNumDomainDims(), 0, domain);
+
+  // Merge and align domain ids of `ret` and ids of `domain`. Domain ids of
+  // `ret` are guranteed to be a subset of ids of `domain`.
```

Comment:
Add some more documentation on why we need to do this and why is the domain a subset.